napari/layers/points/points.py

```diff
@@ -275,6 +279,25 @@ def __init__(
             properties = dataframe_to_properties(properties)
         self._properties = self._validate_properties(properties)
 
+        if default_properties is None:
+            default_properties = {}
+        elif not isinstance(default_properties, dict):
+            default_properties = dataframe_to_properties(default_properties)
+        self._default_properties = default_properties
+
+        if len(data) == 0 and self.properties and not default_properties:
+            raise ValueError(
+                'default_properties must be set for empty Points layers with colors that depend on properties'
+            )
+
+        if (
+            self._default_properties
```

Comment:
```suggestion
            self._default_properties is not None
```

Reply:
(Note that an empty dict evaluates to false! Explicit is better than implicit.)

Reply:
I just saw that you make it an empty dict, above. So, the question becomes, if I pass `properties={'labels': []}` and `default_properties={}`, do we want a KeyError or a ValueError?